lib/matplotlib/quiver.py

```diff
@@ -513,58 +525,233 @@ def _init(self):
+        """
+        self.set_XYUVC(V=V)
+
+    def get_V(self):
+        """Returns the vertical direction components."""
+        return self._V
+
+    def set_C(self, C):
+        """
+        Set the arrow colors.
+
+        Parameters
+        ----------
+        C : array-like
+            The size must the same as the existing X, Y, U, V or be one.
+        """
+        self.set_XYUVC(C=C)
+
+    def get_C(self):
+        """Returns the arrow colors."""
+        return self.get_array()
+
+    def set_XYUVC(self, X=None, Y=None, U=None, V=None, C=None, check_shape=False):
+        """
+        Set the positions (X, Y) and components (U, V) of the arrow vectors
+        and arrow colors (C) values of the arrows.
+        The size of the array must match with existing values. To change
+        the size, all arguments must be changed at once and their size
+        compatible.
+
+        Parameters
+        ----------
+        X, Y : array-like of float or None, optional
+            The arrow locations in the horizontal and vertical directions.
+            Any shape is valid so long as X and Y have the same size.
+        U, V : array-like or None, optional
+            The horizontal and vertical direction components of the arrows.
+            If None it is unchanged.
+            The size must the same as the existing U, V or be one.
+        C : array-like or None, optional
+            The arrow colors. The default is None.
+            The size must the same as the existing U, V or be one.
+        check_shape : bool
+            Whether to check if the shape of the parameters are
+            consistent. Default is False.
+        """
+
+        X = self.get_X() if X is None else X
+        Y = self.get_Y() if Y is None else Y
+        if U is None or isinstance(U, Number):
+            nr, nc = (self._nr, self._nc)
+        else:
+            nr, nc = _extract_nr_nc(U)
+        X, Y = _process_XY(X, Y, nc, nr)
+        N = len(X)
+
         # We need to ensure we have a copy, not a reference
         # to an array that might change before draw().
-        U = ma.masked_invalid(U, copy=True).ravel()
-        V = ma.masked_invalid(V, copy=True).ravel()
-        if C is not None:
-            C = ma.masked_invalid(C, copy=True).ravel()
-        for name, var in zip(('U', 'V', 'C'), (U, V, C)):
-            if not (var is None or var.size == self.N or var.size == 1):
-                raise ValueError(f'Argument {name} has a size {var.size}'
-                                 f' which does not match {self.N},'
-                                 ' the number of arrow positions')
-
+        U = ma.masked_invalid(self._U if U is None else U, copy=True).ravel()
+        V = ma.masked_invalid(self._V if V is None else V, copy=True).ravel()
+        if C is not None or self._C is not None:
+            C = ma.masked_invalid(
+                self._C if C is None else C, copy=True
+            ).ravel()
```

Comment:
This seems extra complicated with the `or` _and_ ternary:
```suggestion
        C = ma.masked_invalid(C, copy=True).ravel() if C is not None else self._C
```

Reply:
But this should be okay to simplify, as `C` is not unmasked later. But actually, I'm a bit confused, as nothing appears to set `self._C` to anything?